src/overlays/actors/ovl_En_Kakasi3/z_en_kakasi3.c

```diff
@@ -21,35 +47,378 @@ const ActorInit En_Kakasi3_InitVars = {
+        func_8010B680(globalCtx, 0x40A5, NULL);
+        func_8002DF54(globalCtx, NULL, 8);
+        this->actionFunc = func_80A91A90;
+    } else if (globalCtx->msgCtx.unk_E3EE == 1) {
+        func_80A90EBC(this, globalCtx, 0);
+        player->stateFlags2 |= 0x800000;
+    }
+}
+
+void func_80A91760(EnKakasi3* this, GlobalContext* globalCtx) {
+
+    func_80A90E28(this);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (this->dialogState == func_8010BDBC(&globalCtx->msgCtx) && func_80106BC8(globalCtx)) {
+        globalCtx->msgCtx.msgMode = 0x37;
+        func_8010BD58(globalCtx, 0x2D);
+        this->actionFunc = func_80A917FC;
+        this->camId = func_800800F8(globalCtx, 0x8E8, -0x63, &this->actor, 0);
+    }
+}
+
+void func_80A917FC(EnKakasi3* this, GlobalContext* globalCtx) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A90E28.s")
+    if (globalCtx->msgCtx.unk_E3EE != 0xF) {
+        func_80A90EBC(this, globalCtx, 1);
+    } else {
+        globalCtx->msgCtx.unk_E3EE = 4;
+        func_80106CCC(globalCtx);
+        func_800803F0(globalCtx, this->camId);
+        this->actionFunc = func_80A911F0;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A90EBC.s")
+void func_80A9187C(EnKakasi3* this, GlobalContext* globalCtx) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A911F0.s")
+    if (func_8010BDBC(&globalCtx->msgCtx) == 5 && func_80106BC8(globalCtx)) {
+        func_80106CCC(globalCtx);
+        func_8010BD58(globalCtx, 0x28);
+        this->actionFunc = func_80A918E4;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A91284.s")
+void func_80A918E4(EnKakasi3* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A91348.s")
+    if (BREG(3) != 0) {
+        // No way!
+        osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ まさか！ ☆☆☆☆☆ %d\n" VT_RST, globalCtx->msgCtx.unk_E3EE);
+    }
+    if ((globalCtx->msgCtx.unk_E3EE == 4 || (globalCtx->msgCtx.unk_E3EE >= 5 && globalCtx->msgCtx.unk_E3EE < 11)) &&
+        (globalCtx->msgCtx.msgMode == 0)) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A915B8.s")
+        func_8010B680(globalCtx, 0x40A6, NULL);
+        this->dialogState = 5;
+        func_800803F0(globalCtx, this->camId);
+        this->camId = -1;
+        func_8002DF54(globalCtx, NULL, 8);
+        this->actionFunc = func_80A91A90;
+    } else if (globalCtx->msgCtx.unk_E3EE == 3 && globalCtx->msgCtx.msgMode == 0) {
+        globalCtx->msgCtx.unk_E3EE = 4;
+        if (BREG(3) != 0) {
+            osSyncPrintf("\n\n");
+            // With this, other guys are OK! That's it!
+            osSyncPrintf(VT_FGCOL(CYAN) "☆☆☆☆☆ これで、他の奴もＯＫ！だ！ ☆☆☆☆☆ %d\n" VT_RST,
+                         globalCtx->msgCtx.unk_E3EE);
+        }
+        this->unk_195 = 1;
```

Comment:
```suggestion
        this->unk_195 = true;
```
Since you're using it as a bool everywhere